pkg/apply/ResourceInfos.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package apply
+
+import (
+	"sort"
+
+	"k8s.io/cli-runtime/pkg/resource"
+)
+
+type ResourceInfos []*resource.Info
+
+var _ sort.Interface = ResourceInfos{}
+
+func (a ResourceInfos) Len() int      { return len(a) }
+func (a ResourceInfos) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a ResourceInfos) Less(i, j int) bool {
+	if !getGvk(a[i]).Equals(getGvk(a[j])) {
+		return getGvk(a[i]).IsLessThan(getGvk(a[j]))
+	}
+	return a[i].String() < a[j].String()
+}
+
+func getGvk(info *resource.Info) Gvk {
+	gvk := info.Object.GetObjectKind().GroupVersionKind()
+	return Gvk{
```

Comment:
Why are we creating a new type for GVK here? It seems unnecessary.

Reply:
Will do